src/MLOps.NET/MLLifeCycleManager.cs

```diff
@@ -71,6 +72,20 @@ public async Task UploadModelAsync(Guid runId, string filePath)
             await ModelRepository.UploadModelAsync(runId, filePath);
         }
 
+        ///<inheritdoc/>
+        public IRun GetBestRun(Guid experimentId, string metricName)
+        {
+            EnsureStorageProviderConfigured();
+            var allRuns = MetaDataStore.GetRuns(experimentId);
+            // Flattening the metrics for all the runs for a given experiment into one list and finding the best among them.
+            var bestRunId = allRuns.SelectMany(r => r.Metrics)
+                .Where(m => m.MetricName.ToLowerInvariant() == metricName.ToLowerInvariant())
+                .OrderByDescending(m => m.Value)
```

Comment:
This assumes that a higher value is better (which is reasonable for e.g. Accuracy). There are however other values, e.g. log loss entropy that we want as small as possible. In 99% of cases we want them maximizes, F1 Score, Recall, Precision, Accuracy, so I think we should leave this as is for now to keep it simple and add support for it later when needed. Just wanted to point it out if it was of interest.

Reply:
This actually will come up sooner or later, let's add an issue?

Reply:
Great idea @shanranm - added #77 to cover this in the future.

Reply:
Thanks @aslotte @shanranm for reviewing it. That was a good catch.